xDAQ15 brings log4cplus 2.0

## Brief summary of issue
Testing compilation of `cmsgemos` on `cc7` with `xdaq15` beta release results in a failure because current `xhal` can't be installed:
```
Error: Package: xhal-3.2.2-1.0.102.dev.4adab12git.centos7.gcc4_8_5.x86_64 (gemos-extras)
           Requires: liblog4cplus-1.2.so.5()(64bit)
```

Compiling current `xhal` fails with errors:
```
src/common/utils/XHALXMLParser.cpp:10:35: error: no matching function for call to ‘log4cplus::Appender::setLayout(std::auto_ptr<log4cplus::Layout>&)’
   myAppender->setLayout( myLayout );
                                   ^
In file included from /opt/xdaq/include/log4cplus/spi/appenderattachable.h:33,
                 from /opt/xdaq/include/log4cplus/logger.h:36,
                 from /home/sturdy/src/sw/xhal/xhalcore/include/xhal/utils/XHALXMLParser.h:39,
                 from src/common/utils/XHALXMLParser.cpp:1:
/opt/xdaq/include/log4cplus/appender.h:216:22: note: candidate: ‘virtual void log4cplus::Appender::setLayout(std::unique_ptr<log4cplus::Layout>)’
         virtual void setLayout(std::unique_ptr<Layout> layout);
                      ^~~~~~~~~
/opt/xdaq/include/log4cplus/appender.h:216:22: note:   no known conversion for argument 1 from ‘std::auto_ptr<log4cplus::Layout>’ to ‘std::unique_ptr<log4cplus::Layout>’
src/common/utils/XHALXMLParser.cpp: In member function ‘boost::optional<xhal::utils::Node> xhal::utils::XHALXMLParser::getNodeFromAddress(uint32_t)’:
src/common/utils/XHALXMLParser.cpp:359:1: warning: no return statement in function returning non-void [-Wreturn-type]
 }
```

### Types of issue
- [x] Compatibility breaking change

## Expected Behavior
Will probably need to set up builds for
* `cc7` with default gcc (4.8.5), for "legacy" (`14.6`) `xdaq`
* `cc7` with  `devtoolset-8` gcc (`8.2.1`), for `xdaq15`
* and when available, the same for `cc8` (not sure what the default compiler will be yet)

## Current Behavior
Unable to compile `xhal` for `xdaq15` testing, and hence unable to compile `cmsgemos`

## Context (for feature requests)
For compatibility with future development.

## Your Environment
`cc7` VM with `xdaq15` prerelease installed (`gemdaq-build-xdaq15`)

--- a/xhalcore/src/common/XHALInterface.cpp
+++ b/xhalcore/src/common/XHALInterface.cpp
@@ -1,5 +1,7 @@
 #include "xhal/XHALInterface.h"
 
+#include <log4cplus/version.h>
+
 int xhal::XHALInterface::index = 0;
 
 xhal::XHALInterface::XHALInterface(const std::string& board_domain_name):
@@ -7,8 +9,11 @@ xhal::XHALInterface::XHALInterface(const std::string& board_domain_name):
   isConnected(false)
 {
   log4cplus::SharedAppenderPtr myAppender(new log4cplus::ConsoleAppender());
-  std::auto_ptr<log4cplus::Layout> myLayout = std::auto_ptr<log4cplus::Layout>(new log4cplus::TTCCLayout());
-  myAppender->setLayout( myLayout );
+#if LOG4CPLUS_VERSION < LOG4CPLUS_MAKE_VERSION(2, 0, 0)
+  myAppender->setLayout(std::auto_ptr<log4cplus::Layout>(new log4cplus::TTCCLayout()));
+#else
+  myAppender->setLayout(std::unique_ptr<log4cplus::Layout>(new log4cplus::TTCCLayout()));
+#endif
   // Following strange construction is required because it looks like log4cplus was compiled withot c++11 support...
   auto t_logger = log4cplus::Logger::getInstance(LOG4CPLUS_TEXT("XHALInterface_"+m_board_domain_name + "_" + std::to_string(index)));
   ++index;

--- a/xhalcore/src/common/utils/XHALXMLParser.cpp
+++ b/xhalcore/src/common/utils/XHALXMLParser.cpp
@@ -1,13 +1,18 @@
 #include "xhal/utils/XHALXMLParser.h"
 
+#include <log4cplus/version.h>
+
 int xhal::utils::XHALXMLParser::index = 0;
 
 xhal::utils::XHALXMLParser::XHALXMLParser(const std::string& xmlFile)
 {
   m_xmlFile = xmlFile;
   log4cplus::SharedAppenderPtr myAppender(new log4cplus::ConsoleAppender());
-  std::auto_ptr<log4cplus::Layout> myLayout = std::auto_ptr<log4cplus::Layout>(new log4cplus::TTCCLayout());
-  myAppender->setLayout( myLayout );
+#if LOG4CPLUS_VERSION < LOG4CPLUS_MAKE_VERSION(2, 0, 0)
+  myAppender->setLayout(std::auto_ptr<log4cplus::Layout>(new log4cplus::TTCCLayout()));
+#else
+  myAppender->setLayout(std::unique_ptr<log4cplus::Layout>(new log4cplus::TTCCLayout()));
+#endif
   auto t_logger = log4cplus::Logger::getInstance(LOG4CPLUS_TEXT("XHALXMLParser_" + std::to_string(index)));
   ++index;
   m_logger = t_logger;

